src/FloatingButton/FloatingButton.tsx

```diff
@@ -0,0 +1,198 @@
+
+  onClick: PropTypes.func,
+
+  offsetX: PropTypes.number,
+  offsetY: PropTypes.number,
+  positionX: PropTypes.oneOf(["left", "center", "right"]),
+  positionY: PropTypes.oneOf(["bottom", "top"]),
+  size: PropTypes.string,
+};
+
+const defaultProps = {
+  animate: true,
+  label: "+",
+  offsetX: 30,
+  offsetY: 30,
+  positionX: "right",
+  positionY: "bottom",
+};
+
+export default class FloatingButton extends React.PureComponent {
+  static propTypes = propTypes;
+  static defaultProps = defaultProps;
+  static cssClass = cssClass;
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      active: false,
+      display: false,
+    };
+  }
+
+  horizontalStyle() {
+    const { offsetX, positionX } = this.props;
+    if (positionX === "center") return {};
+    return positionX === "left"
+      ? {
+          left: `${offsetX}px`,
+        }
+      : {
+          // default to right if a bad positionX input provided
+          right: `${offsetX}px`,
+        };
+  }
+
+  verticalStyle() {
+    const { offsetY, positionY } = this.props;
+    return positionY === "top"
+      ? {
+          top: `${offsetY}px`,
+        }
+      : {
+          // default to bottom orientation
+          bottom: `${offsetY}px`,
+        };
+  }
+
+  mainButtonHandler = () => {
+    const { additionalButtons, animate, onClick } = this.props;
+    const { active } = this.state;
+    if (additionalButtons && additionalButtons.length) {
+      // main button activates additional buttons
+      if (active) {
+        // stop displaying when transition to inactive state complete
+        this.setState({ active: false, display: true });
+        setTimeout(() => this.setState({ display: false }), animate ? 200 : 0);
+      } else {
+        this.setState({ active: true, display: true });
+      }
+    }
+    if (onClick && !active) {
+      onClick();
+    }
+  };
+
+  render() {
+    const {
+      additionalButtons,
+      animate,
+      bgColor,
+      className,
+      closeLabel,
+      label,
+      positionX,
+      positionY,
+      size,
+    } = this.props;
+    const { active, display } = this.state;
+
+    return (
+      <FlexBox
+        className={classnames(
+          cssClass.CONTAINER,
+          cssClass.propStyle(positionX),
+          cssClass.propStyle(positionY),
+          className,
+        )}
+        style={{
+          ...this.horizontalStyle(),
+          ...this.verticalStyle(),
+        }}
+      >
+        <div className={classnames(cssClass.SUB_ELEMENT, className)}>
+          <Button
+            className={cssClass.BUTTON}
+            onClick={this.mainButtonHandler}
+            value={
+              active
+                ? closeLabel || (
+                    <div>
+                      <span className="fa fa-times" /> Close
+                    </div>
+                  )
+                : label
+            }
+            size={size || "regular"}
+            style={{
+              borderRadius: "500px",
```

Comment:
Since this style is always applied, it should just live in the stylesheet. Also, rather than using an arbitrarily large value, I'd just use `border-radius: 50%`

Reply:
I tried to place it in the stylesheet but the styles from the Button component override it. I think that's partially why that component exposes the "style" prop.

Reply:
To your point about border-radius, this is what border-radius: 50% looks like: 
<img width="156" alt="Screen Shot 2019-05-24 at 3 17 31 PM" src="https://user-images.githubusercontent.com/42983512/58357201-232d8e80-7e37-11e9-8dd2-b58a923da6a1.png">

Reply:
Ah super interesting. Looks like percentage values and specific pixel values are applied differently: https://stackoverflow.com/questions/29966499/border-radius-in-percentage-and-pixels-px-or-em